scipy/sparse/csgraph/_flow.pyx

```diff
@@ -668,3 +679,740 @@ cdef ITYPE_t[:] _dinic(
                              capacities, progress, flows, stack):
             pass
     return flows
+
+
+def minimum_cost_flow(csgraph, demand, cost):
+
+    demand = np.asarray(demand)
+    cost = np.asarray(cost)
+    if not isspmatrix_csr(csgraph):
+        raise TypeError("graph must be in CSR format")
+    if not issubclass(csgraph.dtype.type, np.integer):
+        raise ValueError("graph capacities must be integers")
+    elif csgraph.dtype != ITYPE:
+        csgraph = csgraph.astype(ITYPE)
+    if not issubclass(demand.dtype.type, np.integer):
+        raise ValueError("vertex demands must be integers")
+    elif demand.dtype != ITYPE:
+        demand = demand.astype(ITYPE)
+    if not issubclass(cost.dtype.type, np.integer):
+        raise ValueError("edge costs must be integers")
+    elif cost.dtype != ITYPE:
+        cost = cost.astype(ITYPE)
+
+    n_verts = csgraph.indptr.shape[0] - 1
+
+    _network_simplex_checks(csgraph.data, demand, cost)
+    tails = _make_tails(csgraph)
+    row = np.empty(cost.shape, dtype=ITYPE)
+    col = np.empty(cost.shape, dtype=ITYPE)
+    flow_data = np.empty(cost.shape, dtype=ITYPE)
+    result = _network_simplex(csgraph.indices, tails, csgraph.data,
+                              demand, cost, n_verts,
+                              row, col, flow_data)
+    if result[3] == 0:
+        raise ValueError("no flow satisfies all node demands")
+    flow_matrix = csr_matrix((flow_data[0:result[1]],
+                             (row[0:result[1]], col[0:result[1]])),
+                             shape=(n_verts, n_verts))
+    return MinCostFlowResult(result[2], flow_matrix, result[0])
```

Comment:
> Have you chosen to use an array to be able to transfer return values more easily?

Yes. Instead of returning I pass the reference to array and modify them inside the functions. This avoids in deep copying the values while returning them as tuples.

Reply:
I see. Probably, you can have a lightweight Cython extension type wrapping those return values?